src/passes/MergeBlocks.cpp

```diff
@@ -277,6 +277,22 @@ struct MergeBlocks : public WalkerPass<PostWalker<MergeBlocks>> {
   void visitStore(Store* curr) {
     optimize(curr, curr->value, optimize(curr, curr->ptr), &curr->ptr);
   }
+  void visitAtomicRMW(AtomicRMW* curr) {
+    optimize(curr, curr->value, optimize(curr, curr->ptr), &curr->ptr);
+  }
+  // XXX TODO: why doesn't this work for select?
```

Comment:
@kripken the mergeblocks test goes into an infinite loop when I use this for select, but it looks just the same. any ideas?